src/verinfast/agent.py

```diff
@@ -539,7 +547,7 @@ def scanRepos(self):
 
                     os.chdir(curr_dir)
                     if not self.config.dry and self.config.delete_temp:
-                        shutil.rmtree(temp_dir)
+                        shutil.rmtree(temp_dir, onerror=self.handleRemoveReadonly)
```

Comment:
this seems dangerous. Probably best to just error out if we're doing rmtree somewhere we didn't create.

Reply:
The folder is created by the script which I believe is the problem. When Python creates a folder on Windows it is set as Read Only.

I didn't check to see if there was a way to proactively not create a folder as Read Only, which is another possibility.

If you error out, you'd need to create new folders for each repo because you're not going to be able to clear the old ones.